### PR TITLE
feat(plugin-board): add upload snapshot for board 

### LINKED ISSUE #273: add support to store a board snapshot image
A snapshot of a whiteboard will be taken periodically and stored within the board service.  We should add this functionality to board.

### PR DESCRIPTION
**Pull request by tran2**
To add a snapshot for a white board, we upload the snapshot image to spark files then attach metadata to a board.
Resolves #273

### CODE REVIEW

File: src/client/services/board/persistence/persistence.js [hunk 1]
```diff
@@ -65,6 +65,50 @@ var PersistenceService = SparkBase.extend({
...
+            height: image.height || 900,
+            width: image.width || 1600,
+            mimeType: image.type || 'image/png',
+            scr: imageScr.encryptedScr,
+            encryptionKeyUrl: conversation.defaultActivityEncryptionKeyUrl,
+            fileSize: image.size
+          }
+        };
+        return this.spark.request({
+          method: 'PATCH',
```
Comment (ianwremmel) [inline src/client/services/board/persistence/persistence.js]: Why is this a patch? I'm pretty sure POST is the correct verb if I'm understanding the semantics of this action.

File: packages/plugin-board/src/board.js [hunk 2]
```diff
@@ -57,6 +57,47 @@ const Board = SparkPlugin.extend({
...
+            height: image.height || 900,
+            width: image.width || 1600,
+            mimeType: image.type || `image/png`,
+            scr: imageScr.encryptedScr,
+            encryptionKeyUrl: conversation.defaultActivityEncryptionKeyUrl,
+            fileSize: image.size
+          }
+        };
+        return this.spark.request({
+          method: `PATCH`,
```
Comment (ianwremmel) [inline packages/plugin-board/src/board.js]: See question about PATCH/POST

File: src/client/services/board/persistence/persistence.js (hunk 1 above)
Comment (tran2) [inline src/client/services/board/persistence/persistence.js]: I'm doing a PATCH to update the image of the channel (this is specified by the server)
```
{
    "image": {}
}
```
The channel DTO is something like this
```JSON
{
    "channelId": "fd48a240-23eb-11e4-be51-56aa0a6fd5da",
    "channelUrl": "https://board-a.wbx2.com/board/api/v1/channels/fd48a240-23eb-11e4-be51-56aa0a6fd5da",
    "aclUrl": "https://acl-a.wbx2.com/acl/api/v1/acls",
    "parentId": null,
    "type": "board",
    "creatorId": "",
    "createdTime": 1435278171139,
    "properties": { "penColor": "black"},
    "defaultEncryptionKeyUrl" : "", 
    "kmsResourceUrl": "https://encryption-a.wbx2.com/encryption/api/v1/resources/",
    "contentLastUpdatedTime" : 1435278182139,   
    "hiddenSpaceUrl": "https://files-api-a.wbx2.com/v1/spaces/",
    "conversationUrl":null,
    "image":{
          "updatedTime": 1435278182139,
          "updateInterval": 360000,
          "url": "https://files-api-a.wbx2.com/v1/spaces/bytes",
          "height": 1405,
          "width": 1080,
          "fileSize": 4691565,
          "mimeType": "image/png",
          "scr": "",
          "encryptionKeyUrl" : ""
        },
    "aclUrlLink":"https://acl-a.wbx2.com/acl/api/v1/acls",
    "imageUpdateInterval":36000
}
```

File: packages/plugin-board/src/board.js [hunk 3]
```diff
@@ -57,6 +57,47 @@ const Board = SparkPlugin.extend({
...
 
   /**
+   * Adds a snapshot image of the board
+   *
+   * @param {Conversation} conversation - the current conversation that the board belongs
+   * @param {Board~Channel} channel
+   * @param {File} image
+   * @returns {Promise<Board~Channel>}
+   */
+  addSnapshotImage(conversation, channel, image) {
```
Comment (tran2) [inline packages/plugin-board/src/board.js]: I can make the change in the new PR (ACL changes https://github.com/ciscospark/spark-js-sdk/pull/357)

Comment (ianwremmel) [inline packages/plugin-board/src/board.js]: sounds good.

**Comment by tran2** [comment]
@ianwremmel I have renamed the function addSnapshotImage to setSnapshotImage

**Pull request merged**